Danki2/Assets/Scripts/Abilities/AbilityBuilder.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Abilities;
+using Assets.Scripts.Abilities.InstantCast;
+using System;
+using System.Collections.Generic;
+
+namespace Assets.Scripts.Abilities
+{
+    public enum AbilityReference
+    {
+        Slash,
+        ShieldBash,
+    }
+
+    public class AbilityBuilder
+    {
+        private readonly Func<AbilityContext, Ability> buildAction;
+
+        public AbilityBuilder(Func<AbilityContext, Ability> buildAction)
+        {
+            this.buildAction = buildAction;
+        }
+
+        public Ability Build(AbilityContext context)
+        {
+            return this.buildAction(context);
+        }
+
+        // This is where we tie the AbilityReference enum to abilities.
+        private static Dictionary<AbilityReference, AbilityBuilder> abilityBuilders = new Dictionary<AbilityReference, AbilityBuilder>
```

Comment:
Fields should be at the top.

Reply:
I think you're probably right, it would be really nice to keep the static stuff together though. I guess there's an argument for making a new static class for it. What do you think?